aggregator/src/aggregator.rs

```diff
@@ -2751,375 +2647,481 @@ where
+                "tasks/:task_id/reports",
+                upload_cors_preflight,
+            )
+            .put(
+                "tasks/:task_id/aggregation_jobs/:aggregation_job_id",
+                instrumented(api(aggregation_jobs_put::<C>)),
+            )
+            .post(
+                "tasks/:task_id/aggregation_jobs/:aggregation_job_id",
+                instrumented(api(aggregation_jobs_post::<C>)),
+            )
+            .put(
+                "tasks/:task_id/collection_jobs/:collection_job_id",
+                instrumented(api(collection_jobs_put::<C>)),
+            )
+            .post(
+                "tasks/:task_id/collection_jobs/:collection_job_id",
+                instrumented(api(collection_jobs_post::<C>)),
+            )
+            .delete(
+                "tasks/:task_id/collection_jobs/:collection_job_id",
+                instrumented(api(collection_jobs_delete::<C>)),
+            )
+            .post(
+                "tasks/:task_id/aggregate_shares",
+                instrumented(api(aggregate_shares::<C>)),
+            ),
+        StatusCounter::new(&meter),
+    ))
+}
 
-    let aggregator = Arc::new(Aggregator::new(datastore, clock, meter, cfg));
+/// Deserialization helper struct to extract a "task_id" parameter from a query string.
+#[derive(Deserialize)]
+struct HpkeConfigQuery {
+    /// The optional "task_id" parameter, in base64url-encoded form.
+    #[serde(default)]
+    task_id: Option<String>,
+}
 
-    let hpke_config_responding = warp::get()
-        .and(with_cloned_value(Arc::clone(&aggregator)))
-        .and(warp::query::<HashMap<String, String>>())
-        .then(
-            |aggregator: Arc<Aggregator<C>>, query_params: HashMap<String, String>| async move {
-                let hpke_config_bytes = aggregator
-                    .handle_hpke_config(query_params.get("task_id").map(String::as_ref))
-                    .await
-                    .map_err(Arc::new)?;
-                http::Response::builder()
-                    .header(CACHE_CONTROL, "max-age=86400")
-                    .header(CONTENT_TYPE, HpkeConfigList::MEDIA_TYPE)
-                    .body(hpke_config_bytes)
-                    .map_err(|err| {
-                        Arc::new(Error::Internal(format!("couldn't produce response: {err}")))
-                    })
-            },
-        );
+/// API handler for the "/hpke_config" GET endpoint.
+async fn hpke_config<C: Clock>(
+    conn: &mut Conn,
+    State(aggregator): State<Arc<Aggregator<C>>>,
+) -> Result<EncodedBody<HpkeConfigList>, Error> {
+    let query = serde_urlencoded::from_str::<HpkeConfigQuery>(conn.querystring())
+        .map_err(|err| Error::BadRequest(format!("couldn't parse query string: {err}")))?;
+    let hpke_config_list = aggregator
+        .handle_hpke_config(query.task_id.as_ref().map(AsRef::as_ref))
+        .await?;
+
+    conn.headers_mut()
+        .insert(KnownHeaderName::CacheControl, "max-age=86400");
```

Comment:
An option here would be to use [`trillium_caching_headers::CacheControlDirective::MaxAge`](https://docs.rs/trillium-caching-headers/latest/trillium_caching_headers/enum.CacheControlDirective.html)